src/subscript-loc.c

```diff
@@ -207,9 +213,19 @@ r_obj* int_invert_location(r_obj* subscript,
 
     j = -j;
     if (j > n) {
-      struct location_opts updated_opts = *opts;
-      updated_opts.subscript_opts.action = SUBSCRIPT_ACTION_NEGATE;
-      stop_subscript_oob_location(subscript, n, &updated_opts);
+      switch (opts->loc_oob) {
+      case LOC_OOB_REMOVE: {
+        continue;
+      }
```

Comment:
To support this behavior:

``` r
library(dplyr)

df <- tibble(x = 1:5)

slice(df, -6)
#> # A tibble: 5 × 1
#>       x
#>   <int>
#> 1     1
#> 2     2
#> 3     3
#> 4     4
#> 5     5

slice(df, c(-2, -4))
#> # A tibble: 3 × 1
#>       x
#>   <int>
#> 1     1
#> 2     3
#> 3     5
```

<sup>Created on 2022-08-12 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>